phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -2135,7 +2135,7 @@ private PTable createTableInternal(CreateTableStatement statement, byte[][] spli
                 // Upsert physical name for mapped view only if the full physical table name is different than the full table name
                 // Otherwise, we end up with a self-referencing link and then cannot ever drop the view.
                 if (viewType != ViewType.MAPPED
-                        || !physicalNames.get(0).getString().equals(SchemaUtil.getTableName(schemaName, tableName))) {
+                        || !physicalNames.get(0).getString().equals(SchemaUtil.getPhysicalName(schemaName, tableName))) {
```

Comment:
can you include isNamespaceMapped as table name with schema can also be in default namespace.
```
-                        || !physicalNames.get(0).getString().equals(SchemaUtil.getTableName(schemaName, tableName))) {
+                        || !physicalNames.get(0).getString().equals(SchemaUtil.getPhysicalTableName(
+                                SchemaUtil.getTableNameAsBytes(schemaName, tableName), isNamespaceMapped).getNameAsString())) {
```